app/controllers/documents_controller.rb

```diff
@@ -101,7 +101,16 @@ def document_error_messages
   end
 
   def fetch_document
-    @document = document_klass.find(params[:content_id])
+    begin
+      @document = document_klass.find(params[:content_id])
+    rescue Document::RecordNotFound => e
+      flash[:danger] = "Document not found"
+      redirect_to documents_path(document_type: document_type)
```

Comment:
I think doing a 404 here will render the standard Rails one. We want to redirect the User to a sensible place and give them some indication why. I'm not sure if this is standard in other Publishing Apps but this is the same behaviour as master, other than redirecting to the format instead of `manuals_path`.